### DISCUSSION TITLE
DBT snapshot with user defined dbt_valid_from and dbt_valid_to and dbt_updated_at

### DISCUSSION THREAD

**Discussion by ashwintastic** [dbt-labs/dbt-core#9973] [Ideas]
I have already raised a request for this on [#9848](https://github.com/dbt-labs/dbt-core/issues/9848).
My few queries are:
 1. If user-defined column names are not allowed,  **column names** would become dependent on DBT. Imagine in the future, there could be a scenario where we receive an update from DBT stating that from `XYZ` version onwards, DBT doesn't support certain column names (`dbt_valid_from` , `dbt_valid_to`  . This could potentially have a significant impact on the product using. Why a **table column name** is derived from tech-stack?
 2. Is it possible that `dbt_valid_from`, `dbt_valid_to` column names will get support even if they are deprecated ?

**Comment by graciegoheen** [comment]
This is now supported - https://docs.getdbt.com/reference/resource-configs/snapshot_meta_column_names
